src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public class FigureSupplier extends ColorSupplier {
+    public static final int NUMBER_OF_FIGURES_NAME = 5;
+
+    public FigureSupplier() {
+
+    }
+
+    public String getRandomFigureName() {
+        return FigureName.values()[getRandom().nextInt(NUMBER_OF_FIGURES_NAME)].name();
+    }
+
+    public Figure getRandomFigure() {
+        return new Figure(getRandomColor(), getRandomFigureName());

```

Comment:
Instead of directly passing expressions to the `Figure` constructor, create separate variables for the color and figure name for better code readability.